glusterd2/config.go

```diff
@@ -135,11 +155,12 @@ func initConfig(confFile string) error {
 				Error("failed to read config file")
 			return err
 		}
+	} else {
+		config.AddConfigPath("/etc/glusterd2")
+		config.AddConfigPath("/usr/local/etc/glusterd2")
+		config.SetConfigName("glusterd2")
```

Comment:
Conceptually, shouldn't `SetConfigName` occur before `AddConfigPath` ?

Instead of hard-coding, you can derive the config path from `defaultPathPrefix` variable set during build time. We do not want those to pick up files from either of these.

Reply:
default config file was removed previously. are we introducing that again?

Reply:
Ah, that's a good point @aravindavk 

I'd prefer that we don't. The `--config` shall be explicit argument in systemd service file anyway.

Reply:
The order of `SetConfigName`, `AddConfigPath` or `SetConfigType` doesn't matter until `ReadInConfig` or `MergeInConfig` is called (which I've missed for the default case).

About having default search paths. I don't think there's any harm in having it. If the systemd-unit specifies a config file, we use it. And as we don't merge config files, there's no harm.

I'll correct the hardcoding of the paths.